main.js

```diff
@@ -616,3 +627,60 @@ function prevb(){
 grab("fl").onclick = flashlight;
 grab("prevb").onclick = prevb;
 grab("nextb").onclick = nextb;
+
+/* removes all courses from the courses list. */
+function rmcourses(){
+	grabclass("courses")[0].innerHTML = '<div style="text-align: center; margin: 0.8em;"> current courses </div>';
+	courses = [];
+}
+
+function idiot(isLoad){
+	isLoad = typeof isLoad !== 'undefined' ? isLoad : false;
+	var tempstate = getstate();
+	if(tempstate !== {}) {
+		tempstate.i = 1;
+	}
+	if(!isLoad) {
+		rmcourses();
+		rmblocks();
+		courses = [];
+	}
+	dontfill_color = "#000";
+	palette = ["#CC0000", "#7A00CC", "#29A329", "#CCCC00",  "#00CCCC", "#00008A", "#002900", "#E62EB8", "#005C5C", "#CC3300", "#808080", "#00FF00", "#666633", "#002E2E"];
+	var all = document.querySelectorAll("*");
+	for(i=0; i<all.length; i++) {
+		all[i].classList.add("idiot");
+	}
+	isidiot = 1;
+	document.getElementById('idiot-link').innerHTML = 'take me back!';
+	if(!isLoad) {
+		restorestate(tempstate);
+	}
+}
+
+function clever(){
+	var tempstate = getstate();
+	if(tempstate !== {}) {
+		tempstate.i = 0;
+	}
+	rmcourses();
+	rmblocks();
+	courses = [];
+	palette = ["#fcdfdf", "#fcebdf", "#dffce1", "#dffcfa", "#dff3fc", "#dfe6fc", "#e4dffc", "#f0dffc"];
+	dontfill_color = "#ddd";
+	var all = document.querySelectorAll("*");
+	for(i=0; i<all.length; i++) {
+		all[i].classList.remove("idiot");
+	}
+	isidiot = 0;
+	document.getElementById('idiot-link').innerHTML = 'i am an idiot!';
+	restorestate(tempstate);
+}
+
+grab("idiot-link").onclick = function(){ if(isidiot === 1) { clever();} else { idiot();} };
+
+function idiotCheck() {
```

Comment:
eh I missed this. the right way to do this is via a [CSS media query](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Targeting_media_features). don't touch the JS, go in the CSS:
```css
@media(max-width: 768px){
    #idiot-link { display: none }
}
```

Reply:
Done, thanks for the info. I didn't know about that media queries ^^